diff --git a/src/video/cocoa/SDL_cocoametalview.m b/src/video/cocoa/SDL_cocoametalview.m
index af84e935..69e2a808 100644
--- a/src/video/cocoa/SDL_cocoametalview.m
+++ b/src/video/cocoa/SDL_cocoametalview.m
@@ -71,7 +71,15 @@ static bool SDLCALL SDL_MetalViewEventWatch(void *userdata, SDL_Event *event)
  */
 - (CALayer *)makeBackingLayer
 {
-    return [self.class.layerClass layer];
+    CAMetalLayer *metalLayer = [self.class.layerClass layer];
+
+    // Allow CA to resize the layer properly
+    metalLayer.autoresizingMask = kCALayerWidthSizable | kCALayerHeightSizable;
+
+    // Important for live-resize: redraw instead of stretching old frames
+    metalLayer.needsDisplayOnBoundsChange = YES;
+
+    return metalLayer;
 }
 
 - (instancetype)initWithFrame:(NSRect)frame
@@ -88,6 +96,13 @@ static bool SDLCALL SDL_MetalViewEventWatch(void *userdata, SDL_Event *event)
         // Allow resize.
         self.autoresizingMask = NSViewWidthSizable | NSViewHeightSizable;
 
+        // We want to redraw the layer during live window resize.
+        self.layerContentsRedrawPolicy = NSViewLayerContentsRedrawDuringViewResize;
+
+        // Not strictly necessary, but in case something goes wrong with live window
+        // resize, this layer placement makes it more obvious what's going wrong.
+        self.layerContentsPlacement = NSViewLayerContentsPlacementTopLeft;
+
         self.layer.opaque = opaque;
 
         SDL_AddWindowEventWatch(SDL_WINDOW_EVENT_WATCH_EARLY, SDL_MetalViewEventWatch, (__bridge void *)(self));
